tests/testthat/test_reset.R

```diff
@@ -2,12 +2,281 @@ context("reset.R functions")
 
 
 test_that('reset.R: ', {
-    # testing valid t_obj
-    new <- resetDGEobj(t_obj)
-    expect_s3_class(new, "DGEobj")
-    expect_setequal(names(new), c("counts_orig", "counts", "design_orig", "design",
-                                  "geneData_orig", "geneData", "granges_orig", "granges"))
-    expect_equal(dim(new), c(1000, 48))
+
+    test_t_obj  <- t_obj
+    test_t_reset <- resetDGEobj(t_obj)
+
+    # The following are defined in setup.R: t_obj, t_isoform_obj,t_exon_obj, t_protein_obj
+
+    # test gene level data
+    test_gene_obj <- function(gene_data) {
```

Comment:
Instead of making these "functions" you just make them different "tests" - so you would have a test_that('Reset.R: gene level data', { <your test code here> })  They go into "test_that function calls, instead of your own defined function calls.  This eliminates the awkward "expect_success" at the end which won't be needed.  This also means you'll get a report of all the issues in the function, instead of just the details of the success call.